src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,222 @@
+                + "AS manufacturer_country FROM cars JOIN manufacturers "
+                + "ON cars.manufacturer_id = manufacturers.id "
+                + "WHERE cars.id = ? AND cars.deleted = FALSE;";
+        Car currentCar = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getCarStatement = connection.prepareStatement(getCarQuery)) {
+            getCarStatement.setLong(1, id);
+            ResultSet dataFromDB = getCarStatement.executeQuery();
+            if (dataFromDB.next()) {
+                currentCar = parseCarByDataFromDB(dataFromDB);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get Car object with input ID: " + id, e);
+        }
+        if (currentCar != null) {
+            currentCar.setDrivers(getDriversForCurrentCar(id));
+        }
+        return Optional.ofNullable(currentCar);
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String getAllCarsQuery = "SELECT cars.id AS car_id, model, "
+                + "manufacturers.id AS manufacturer_id,"
+                + " manufacturers.name AS manufacturer_name, manufacturers.country "
+                + "AS manufacturer_country FROM cars JOIN manufacturers "
+                + "ON cars.manufacturer_id = manufacturers.id WHERE cars.deleted = FALSE;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                Statement getAllCarsStatement = connection.createStatement()) {
+            ResultSet dataFromDB = getAllCarsStatement.executeQuery(getAllCarsQuery);
+            while (dataFromDB.next()) {
+                cars.add(parseCarByDataFromDB(dataFromDB));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't select all cars from DB", e);
+        }
+        for (Car car : cars) {
+            car.setDrivers(getDriversForCurrentCar(car.getId()));
+        }
+        return cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String updateCarQuery = "UPDATE cars SET model = ?, manufacturer_id = ? "
+                + "WHERE id = ? AND deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateCarStatement = connection
+                        .prepareStatement(updateCarQuery)) {
+            updateCarStatement.setString(1, car.getModel());
+            updateCarStatement.setLong(2, car.getManufacturer().getId());
+            updateCarStatement.setLong(3, car.getId());
+            updateCarStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update Car object by input data: " + car, e);
+        }
+        deleteAllRelationBetweenCurrentCarAndDrivers(car.getId());
+        if (car.getDrivers() != null) {
+            insertDrivers(car);
+        }
```

Comment:
```suggestion
        deleteAllRelationBetweenCurrentCarAndDrivers(car.getId());
        insertDrivers(car);
```
if app structure is correct we can expect that drivers won't be null

Reply:
Agree